tpke/src/api.rs

```diff
@@ -4,144 +4,108 @@
 
 // TODO: Refactor this module to deduplicate shared code from tpke-wasm and tpke-wasm.
 
-use std::convert::TryInto;
-
-use ark_ec::{AffineCurve, ProjectiveCurve};
-use ark_ff::{BigInteger256, ToBytes};
-
-// Fixing some of the types here on our target engine
-// TODO: Consider fixing on crate::api level instead of bindings level
-type E = ark_bls12_381::Bls12_381;
-type TpkePublicKey = ark_bls12_381::G1Affine;
-type TpkePrivateKey = ark_bls12_381::G2Affine;
-type TpkeCiphertext = crate::Ciphertext<E>;
-type TpkeDecryptionShare = crate::DecryptionShareFast<E>;
-type TpkePublicDecryptionContext = crate::PublicDecryptionContextFast<E>;
-type TpkeSharedSecret =
-    <ark_bls12_381::Bls12_381 as ark_ec::PairingEngine>::Fqk;
-
-#[derive(Clone, Debug)]
-pub struct PrivateDecryptionContext {
-    pub b_inv: ark_bls12_381::Fr,
-    pub decrypter_index: usize,
+use ark_serialize::*;
+
+pub type E = ark_bls12_381::Bls12_381;
+pub type TpkeDkgPublicKey = ark_bls12_381::G1Affine;
+pub type TpkeG1Prepared = <E as ark_ec::PairingEngine>::G1Prepared;
+pub type TpkeG1Affine = <E as ark_ec::PairingEngine>::G1Affine;
+pub type TpkePrivateKey = ark_bls12_381::G2Affine;
+pub type TpkeUnblindingKey = ark_bls12_381::Fr;
+pub type TpkeDomainPoint = ark_bls12_381::Fr;
+pub type TpkeCiphertext = crate::Ciphertext<E>;
+pub type TpkeDecryptionShareSimplePrecomputed =
+    crate::DecryptionShareSimplePrecomputed<E>;
+pub type TpkeDecryptionShareSimple = crate::DecryptionShareSimple<E>;
+pub type TpkePublicDecryptionContext = crate::PublicDecryptionContextSimple<E>;
+pub type TpkeSharedSecret = <E as ark_ec::PairingEngine>::Fqk;
+pub type TpkeResult<T> = crate::Result<T>;
+pub type TpkePrivateDecryptionContext =
+    crate::PrivateDecryptionContextSimple<E>;
+
+pub fn encrypt(
+    message: &[u8],
+    aad: &[u8],
+    public_key: &TpkeDkgPublicKey,
+) -> Ciphertext {
+    // TODO: Should rng be a parameter?
```

Comment:
why?

Reply:
`tpke` is a library and `api.rs` is supposed to be used in bindings. So maybe the library user/bindings user should have the choice of managing randomness themselves. Should the cryptographic library ship with a baked-in source of randomness or should that be left to the user? What if the user of the binding wants to test this library in a controlled environment (with controlled randomness, i.e. a hardcoded seed)?